test/KetlAttestation.deploy.ts

```diff
@@ -0,0 +1,185 @@
+import { BigNumber, providers } from 'ethers'
+import { DEV_KETL_ATTESTATION_CONTRACT } from '@big-whale-labs/constants'
+import { KetlAttestation, KetlAttestation__factory } from '../typechain'
+import { ethers, upgrades } from 'hardhat'
+import { expect } from 'chai'
+import { getFakeAttestationCheckerVerifier } from './utils/fakes'
+import {
+  getLegacyMintCalldata,
+  getLegacyRegisterEntanglementCalldata,
+  getLegacyTokenHolders,
+} from '../scripts/helpers'
+
+describe('KetlAttestation Deploy Tests', () => {
+  it('should be able to deploy KetlAttestation and populate data correctly', async function () {
+    const uri = 'https://game.example/api/item/{id}.json'
+    const version = '0.0.1'
+
+    const realProvider = new providers.JsonRpcProvider(
+      process.env.ETH_RPC,
+      'maticmum'
+    )
+    const realKetlAttesation = await KetlAttestation__factory.connect(
+      DEV_KETL_ATTESTATION_CONTRACT,
+      realProvider
+    )
+    const attestorPublicKey = await realKetlAttesation.attestorPublicKey()
+
+    this.signers = await ethers.getSigners()
+    this.owner = this.signers[0]
+    this.user = this.signers[1]
+
+    this.poseidonT3Factory = await ethers.getContractFactory('PoseidonT3')
+    this.poseidonT3 = await this.poseidonT3Factory.deploy()
+    await this.poseidonT3.deployed()
```

Comment:
Maybe we can extract this into `deployPoseidonT3`?

Reply:
You need only address as I can see

Reply:
I think this makes sense, but I'd rather do it as part of another PR focused on figuring out we want to structure tests for this repo in general. I just copied the code over from the other test file for convenience.